File: algorithms/interview-preparation-kit-2/js/code-response/hackerrankInString/hackerrankInString.spec.mjs
import { hackerrankInString } from './hackerrankInString.mjs';

describe('Hackerrank In String', () => {
  test('hackerrank exist in hereiamstackerrank', () => {
    expect(hackerrankInString('hereiamstackerrank')).toEqual('YES');
  });

  test('hackerrank exist in hackerworld', () => {
    expect(hackerrankInString('hackerworld')).toEqual('NO');
  });

  test('hackerrank exist in hhaacckkekraraannk', () => {
    expect(hackerrankInString('hhaacckkekraraannk')).toEqual('YES');
  });

  test('hackerrank exist in rhbaasdndfsdskgbfefdbrsdfhuyatrjtcrtyytktjjt', () => {
    expect(hackerrankInString('rhbaasdndfsdskgbfefdbrsdfhuyatrjtcrtyytktjjt')).toEqual('NO');
  });
});

File: algorithms/interview-preparation-kit-2/ruby/code-response/hackerrankInString/hackerrankInString.spec.rb
require_relative "./hackerrankInString"

describe "Hackerrank In String" do
  context "hackerrank exist in hereiamstackerrank" do
    it "YES" do
      expect(hackerrankInString('hereiamstackerrank')).to eq('YES')
    end
  end    
end

describe "Hackerrank In String" do
  context "hackerrank exist in hackerworld" do
    it "NO" do
      expect(hackerrankInString('hackerworld')).to eq('NO')
    end
  end    
end

describe "Hackerrank In String" do
  context "hackerrank exist in hhaacckkekraraannk" do
    it "YES" do
      expect(hackerrankInString('hhaacckkekraraannk')).to eq('YES')
    end
  end    
end

describe "Hackerrank In String" do
  context "hackerrank exist in rhbaasdndfsdskgbfefdbrsdfhuyatrjtcrtyytktjjt" do
    it "NO" do
      expect(hackerrankInString('rhbaasdndfsdskgbfefdbrsdfhuyatrjtcrtyytktjjt')).to eq('NO')
    end
  end    
end
